cmd/bootkube/render.go

```diff
@@ -111,12 +116,39 @@ func flagsToAssetConfig() (c *asset.Config, err error) {
 			return nil, err
 		}
 	}
+
+	_, podNet, err := net.ParseCIDR(renderOpts.podCIDR)
+	if err != nil {
+		return nil, err
+	}
+
+	_, serviceNet, err := net.ParseCIDR(renderOpts.serviceCIDR)
+	if err != nil {
+		return nil, err
+	}
+
+	if podNet.Contains(serviceNet.IP) || serviceNet.Contains(podNet.IP) {
+		return nil, fmt.Errorf("Pod CIDR %s and service CIDR %s must not overlap", podNet.String(), serviceNet.String())
+	}
+
+	dnsServiceIP, err := pickDNSServiceIP(renderOpts.serviceCIDR)
+	if err != nil {
+		return nil, err
+	}
+
+	if !renderOpts.selfHostKubelet {
+		fmt.Fprintf(os.Stderr, "Host kubelet MUST use --cluster-dns=%s\n", dnsServiceIP)
```

Comment:
I've updated the examples in the PR description. The message should only be shown if its non-default now.